### DISCUSSION TITLE
Adding margin/shift to pre-defined constraint

### DISCUSSION THREAD

**Discussion by mfry-ll** [yalmip/YALMIP#1109] [Optimization modelling] [answered]
Is there a way to add a margin to an already defined constraint?  I'm thinking along the lines of the following:
```
x = sdpvar(2);
ct = x >= 0; % A positive semidefinite matrix
ct = shift(ct, 1e-7) % Equivalent to ct = x >= 1e-7 * eye(2)
optimize(ct, x(1,1))
```
Obviously I could hard code the shift when generating the constraint, but being able to shift the constraint AFTER generating it is very helpful.

**Comment by johanlofberg** [comment, marked as answer]
No (well yes but it would be more cumbersome than simply adding the shifted constraint instead to a model containing everything but this constraint). If you want to try different margins etc, define an optimizer object with the margin as a parameter

**Reply by mfry-ll** [reply]
Hmm, I'm not sure if making the shift an `sdpvar` will work in my case...I'll tinker around.  Thanks!

**Reply by mfry-ll** [reply]
What does the more cumbersome approach look like? It may help me avoid restructuring code I want to keep relatively static

**Reply by johanlofberg** [reply]
something like 

```
thisConstraints = MyConstraints(thisindex)
MyConstraint(thisindex) = sdpvar(thisConstraint) >= thismargin;
```

However, the example you showed would work with optimizer without issues so I don't see why it wouldn't work in your case

**Reply by mfry-ll** [reply]
yes, that works great! the problem with the optimizer route is that a [function in a toolbox](https://github.com/iqcToolbox/iqcToolbox/blob/master/src/functions/backend/kypLmiLti.m) accepts only the decision variables from the original constraint, and changing the signature to also treat a decision variable on the shift would affect downstream functions.
